File: src/IsbInterop.CSharp/Presentation/IAction.cs
using System;

namespace IsbInterop.Presentation
{
  /// <summary>
  /// Действие формы.
  /// </summary>
  public interface IAction : IBaseIsbObject
  {
    /// <summary>
    /// Выполняет действие.
    /// </summary>
    /// <returns>True, если действие было выполнено, иначе false.</returns>
    bool Execute();

    /// <summary>
    /// Выполняет действие.
    /// </summary>
    /// <param name="timeout">Таймаут.</param>
    /// <returns>True, если действие было выполнено, иначе false.</returns>
    bool Execute(TimeSpan timeout);
  }
}

File: src/IsbInterop/Presentation/IAction.vb
Namespace Presentation

  ''' <summary>
  ''' Действие формы.
  ''' </summary>
  Public Interface IAction
    Inherits IIsbComObjectWrapper

    ''' <summary>
    ''' Выполняет действие.
    ''' </summary>
    ''' <returns>True, если действие было выполнено, иначе false.</returns>
    Function Execute() As Boolean

    ''' <summary>
    ''' Выполняет действие.
    ''' </summary>
    ''' <param name="timeout">Таймаут.</param>
    ''' <returns>True, если действие было выполнено, иначе false.</returns>
    Function Execute(timeout As TimeSpan) As Boolean
  End Interface
End Namespace